src/pages/NewPost/NewPostTabs.tsx

```diff
@@ -1,27 +1,117 @@
+  MAXARTICLELENGTH,
+} from 'components/Posts/PostsMaker/PostMakerContainer/PostMakerContainer';
+import CKeditorMaker from 'components/Posts/PostsMaker/PostMakerContent/CKeditorMaker';
+import PollPostMaker from 'components/Posts/PostsMaker/PostMakerContent/PollPostMaker';
+import {
+  IpostPreviewSubmission,
+  IpostContentType,
+  IpostArticle,
+  IpollOption,
+  IpostPoll,
+} from 'types/types';
 
 type TnewPostTabs = {
   cId: string;
 };
 
 export default function NewPostTabs({ cId }: TnewPostTabs): JSX.Element {
-  const [currentTab, setCurrentTab] = useState(0);
-
   const tabs = [
-    { name: 'Post' },
-    { name: 'Poll' },
-    { name: 'Bounty' },
-    { name: 'Vote' },
+    { name: 'post' },
+    { name: 'poll' },
+    { name: 'bounty' },
+    { name: 'vote' },
   ];
+
+  const userInfo = useRecoilValue(userInfoAtom);
+  const [currentTab, setCurrentTab] = useState(0);
+
+  const [postPreviewSubmission, setPostPreviewSubmission] =
+    useState<IpostPreviewSubmission>({
+      postType: 'article',
+      accessTokenId: cId,
+      accessMinimumTokenBalance: 0,
+      authorUserId: userInfo?.uid ?? '',
+      authorUserName: userInfo?.uid ?? '',
+      authorPublicKey: userInfo?.uid ?? '',
+      authorProfileImageUrl: userInfo?.uid ?? '',
+      title: '',
+      creationDate: new Date().getTime(),
+      category: '',
+      upVoteUserIds: [],
+      downVoteUserIds: [],
+      numberOfComments: 0,
+    });
+
+  // Post content states
+  const [richTextContent, setRichTextContent] = useState('');
+
+  const [articleContent, setArticleContent] = useState<IpostArticle>({
+    postType: 'article',
+    content: richTextContent,
+  });
+
+  const [pollOptions, setPollOptions] = useState<Array<IpollOption>>(() => [
+    { id: Date.now().toString(), name: '', voteUserIds: [] },
+  ]);
+  const [pollContent, setPollContent] = useState<IpostPoll>({
+    postType: 'poll',
+    content: '',
+    options: pollOptions,
+  });
+
+  useEffect(() => {
+    setPollContent((prevState) => ({
+      ...prevState,
+      options: pollOptions,
+    }));
+  }, [pollOptions]);
+
+  useEffect(() => {
+    setPollContent((prevState) => ({
+      ...prevState,
+      content: richTextContent,
+    }));
+    setArticleContent((prevState) => ({
+      ...prevState,
+      content: richTextContent,
+    }));
+  }, [richTextContent]);
+
+  function getPostContent(): IpostContentType {
+    switch (postPreviewSubmission.postType) {
+      case 'poll':
+        return pollContent;
+      default:
+        return articleContent;
+    }
+  }
+
+  function switchPostContentType(tabId: number) {
+    setCurrentTab(tabId);
+
+    if (tabs[tabId].name === 'post') {
+      setPostPreviewSubmission((prevState) => ({
+        ...prevState,
+        postType: 'article',
+      }));
+    }
+    if (tabs[tabId].name === 'poll') {
+      setPostPreviewSubmission((prevState) => ({
+        ...prevState,
+        postType: 'poll',
+      }));
+    }
```

Comment:
I think lines 99-110 can be summarized:

      setPostPreviewSubmission((prevState) => ({
        ...prevState,
        postType: tabs[tabId].name === 'poll' ? 'poll' : article,
      }));

Reply:
don t we have votes incoming?

Reply:
fixed